addons/test_mail/tests/test_mail_followers.py

```diff
@@ -417,62 +414,154 @@ def test_auto_subscribe_notify_email(self):
-             ('message_type', '=', 'user_notification')
+            ('res_id', '=', test.id),
+            ('model', '=', 'mail.test.track'),
+            ('message_type', '=', 'user_notification')
         ])
         notif = self.env['mail.notification'].search([
             ('mail_message_id', '=', mail_message.id),
-            ('res_partner_id', '=', common_partner.id)
+            ('res_partner_id', '=', self.common_partner.id)
         ])
         self.assertEqual(len(notif), 1)
         self.assertEqual(notif.notification_type, 'email')
 
         subtype = self.env.ref('mail.mt_comment')
-        res = self.env['mail.followers']._get_recipient_data(test, 'comment', subtype.id, pids=common_partner.ids)
-        partner_notif = [r for r in res if r[0] == common_partner.id]
+        res = self.env['mail.followers']._get_recipient_data(test, 'comment', subtype.id, pids=self.common_partner.ids)
+        partner_notif = [r for r in res if r[0] == self.common_partner.id]
         self.assertEqual(len(partner_notif), 1)
         self.assertEqual(partner_notif[0][3], 'email')
 
-@tagged('post_install', '-at_install')
-class UnlinkedNotificationTest(TestMailCommon):
-    def test_unlinked_notification(self):
-        """
-        Check that we unlink the created user_notification after unlinked the related document
-
-        Post install because we need the registery to be ready to send notification
-        """
-        common_partner = self.env['res.partner'].create({"name": "demo1", "email": "demo1@test.com"})
-        user_1 = self.env['res.users'].create({'login': 'demo1', 'partner_id': common_partner.id, 'notification_type': 'inbox'})
-
-        test = self.env['mail.test.track'].create({"name": "Test Track", "user_id": user_1.id})
-        test_id = test.id
+    @users('employee')
+    def test_notification_unlink(self):
+        """ Check that we unlink the created user_notification after unlinked the
+        related document. Post install because we need the registry to be ready
+        to send notifications """
+        test = self.env['mail.test.track'].create({"name": "Test Track", "user_id": self.user_1.id})
         mail_message = self.env['mail.message'].search([
-             ('res_id', '=', test_id),
-             ('model', '=', 'mail.test.track'),
-             ('message_type', '=', 'user_notification')
+            ('res_id', '=', test.id),
+            ('model', '=', 'mail.test.track'),
+            ('message_type', '=', 'user_notification')
         ])
         self.assertEqual(len(mail_message), 1)
         test.unlink()
         mail_message = self.env['mail.message'].search([
-             ('res_id', '=', test_id),
-             ('model', '=', 'mail.test.track'),
-             ('message_type', '=', 'user_notification')
+            ('res_id', '=', test.id),
+            ('model', '=', 'mail.test.track'),
+            ('message_type', '=', 'user_notification')
         ])
         self.assertEqual(len(mail_message), 0)
```

Comment:
Could be a search_count for both assertions maybe.
Domain is also common so it could be a variable?